packages/Uploader/src/Uploader.js

```diff
@@ -0,0 +1,221 @@
+    files has been correctly uploaded, only means that they were processed, to find about the state of each file
+    iterate over the files list.
+  */
+  onCompleted: PropTypes.func,
+  /**
+    you can pass an array of header objects as need it ex.
+    <Uploader headers={[{ "API-Key": "your-api-key" }, { Accept: "application/json" }, { "X-CSRF-Token": "your-token"} ]} ...>
+      {({...}) => <YourUI />}
+    </Uploader>
+  */
+  headers: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+  a11yText: null,
+  okFileTypes: ["*/*"],
+  canChooseMultiple: true,
+  defaultIsDisabled: false,
+  hasAutoUpload: true,
+  headers: [],
+  isBodyDroppable: true,
+  maxFileSize: oneMebibyte * 10, // 1048576bytes * 10 = 10,485,760 Mebibytes
+  onChange: () => {},
+  onCompleted: () => {},
+};
+
+function getDocumentBody() {
+  return document.body;
+}
+
+function getContainer(refContainer) {
+  return function findContainer() {
+    return refContainer.current;
+  };
+}
+
+const Uploader = React.forwardRef((props, ref) => {
+  const {
+    a11yText,
+    okFileTypes,
+    canChooseMultiple,
+    hasAutoUpload,
+    maxFileSize,
+    onChange,
+    onCompleted,
+    isBodyDroppable,
+    endpoint,
+    children,
+    defaultIsDisabled,
+    headers,
+  } = props;
+
+  const refInput = React.useRef();
+  const refContainer = React.useRef(null);
+  const refId = React.useRef(uuidv4()).current;
+  const i18n = useI18n();
+  const label = a11yText || i18n.t("uploader.label");
+
+  React.useImperativeHandle(ref, () => ({
+    focus: () => {
+      refInput.current.focus();
+    },
+  }));
+
+  const { files, isCompleted, isDisabled, removeFile, cancelFile, setFiles, upload } = useProcessFiles({
+    defaultIsDisabled,
+    endpoint,
+    hasAutoUpload,
+    headers,
+    onChange,
+    onCompleted,
+  });
+
+  const handleChange = React.useCallback(
+    function handleChange(event) {
+      if (isDisabled) return;
+
+      const files = getFiles({ event, maxFileSize, okFileTypes, endpoint });
+      setFiles(() => {
+        if (refInput.current) {
+          refInput.current.value = "";
+        }
+        return canChooseMultiple ? files : [files[0]]; // in case only allow one file per upload
+      });
+    },
+    [canChooseMultiple, endpoint, isDisabled, maxFileSize, okFileTypes, setFiles]
+  );
+
+  const { isDragLeave, isDraggingOver } = useDragAndDropEvents({
+    dropArea: isBodyDroppable ? getDocumentBody : getContainer(refContainer),
+    handleChange,
+    defaultIsDisabled,
+  });
+
+  const value = React.useMemo(() => {
+    function FileInput(props) {
+      const { children } = props;
+      return (
+        <div css={containerStyles} ref={refContainer}>
+          <input
+            multiple={canChooseMultiple}
+            id={refId}
+            onChange={handleChange}
+            ref={refInput}
+            type="file"
+            accept={okFileTypes.join(",")}
+          />
+          <label htmlFor={refId}>
```

Comment:
you could avoid the logic of generating an id by simply using `<input aria-label={label} />` and removing the actual `<label>`.